build-a-brain/index.rst

```diff
@@ -0,0 +1,197 @@
+(insofar as they can be 'dereferenced'
+to access large amounts of information
+which they do not directly carry).
+However, they are 'semantic'
+(unlike pointers in computer science)
+because these representations capture relations
+in a semantic vector space
+in virtue of their distances to one another,
+as typically envisaged by connectionists.
+
+The four main topics in the book
+that are used to describe the architecture
+are semantics, syntax, control, and learning and memory.
+The discussion of semantics considers
+how semantic pointers are generated from information
+that impinges on the senses,
+reproducing details of
+the spiking tuning curves in the visual system.
+Syntax is captured by demonstrating
+how very large structures can be encoded,
+by binding semantic pointers to one another.
+The section on control considers
+the role of the basal ganglia
+and other structures in routing information
+throughout the brain to control cognitive tasks.
+The section on learning memory describes
+how the SPA includes adaptability
+(despite this not being a focus of the `NEF`_)
+showing how detailed spiking timing during learning
+can be incorporated into the basal ganglia model
+using a biologically plausible STDP-like rule.
+
+An example
+==========
+
+In chapter 7 of the book,
+the SPA Unified Network (Spaun) model is presented
+that demonstrates how
+a wide variety of (about 8) cognitive and non-cognitive tasks
+can be integrated in a single large-scale, spiking neuron model.
+Spaun switches tasks and provides all responses
+without any manual change in parameters from a programmer.
+Essentially, it is a fixed model
+that integrates perception, cognition, and action
+across several different tasks
+(many of which are described elsewhere on this site).
+Spaun is the most complex example
+of an SPA model to date (see figure 1).
+
+.. figure:: spaun-1.png
+   :align: center
+
+   Figure 1: A high-level depiction of the Spaun model,
+   with all of the central features
+   of a general Semantic Pointer Architecture.
+   Visual and motor hierarchies provide semantics
+   via connections to natural input (images)
+   and output (a nonlinear dynamical arm model).
+   Central information manipulation
+   depends on syntactic structure for several tasks.
+   Control is largely captured by
+   the basal ganglia action selection elements.
+   Memory and learning take place in both basal ganglia and cortex.
+   The model itself consists of just under a million spiking neurons.
+
+Let us consider how the model would run
+on one run of one cognitive task
+(`see a video of Spaun performing this task <spaunvideos>`_).
+This task is analogous to the
+`>Raven's Matrix task
+<http://en.wikipedia.org/wiki/Raven's_Progressive_Matrices>`_,
+which requires people to figure out a pattern in the input,
+and apply that pattern to new input to produce novel output.
+For instance given the following input
+"[1] [11] [111] [2] [22] [222] [3] [33] ?"
```

Comment:
If we're going to change this string of characters, I would probably want to change all of the characters that correspond to Spaun inputs... and I'm not sure if MathJax or `monospace` would look better? If it's ok with you, let's copy this over as it was on nengo.ca first and fix up the formatting later?